src/selection_criteria.rs

```diff
@@ -304,7 +305,7 @@ impl ReadPreference {
         let mut doc = doc! { "mode": mode };
 
         if let Some(max_stale) = max_staleness {
-            doc.insert("maxStalenessSeconds", max_stale.as_secs());
+            doc.insert("maxStalenessSeconds", max_stale.as_secs().try_into().unwrap_or(i64::MAX));
```

Comment:
Why use a default value here rather than return an error?

Reply:
My original thought was that for the purposes of `maxStalenessSeconds`, `i64::MAX` is effectively "forever", but you're probably right that we should be more precise and error here for invalid values. Updated.